tests/test_llm.py

```
@@ -411,6 +411,132 @@ async def test_invoke_command_no_args(self, mock_run: AsyncMock) -> None:
+        from sova.llm.models import BatchRequest, BatchResult, LLMResult
+
+        req = BatchRequest(custom_id="req-1", prompt="hello")
+        mock_result = [
+            BatchResult(
+                request=req,
+                result=LLMResult(text="sequential response", model="sonnet"),
+            )
+        ]
+
+        with patch("sova.llm.providers.anthropic_batch.create_batch_provider", return_value=None):
+            with patch("sova.llm.client.get_provider") as mock_get_provider:
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)